File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int,int> w,l;
        for(int i=0;i<matches.size();i++)
        {
            if(w[matches[i][0]]!=-1)
            {
                w[matches[i][0]]++;
                w[matches[i][1]]=-1;
                l[matches[i][1]]++;
            }
            else
            {
                // l[matches[i][0]]++;
                l[matches[i][1]]++;
                w[matches[i][1]]=-1;
            }
        }
        
        // for(auto x:w)cout<<x.first<<" "<<x.second<<endl;
        // cout<<endl;
        // for(auto x:l)cout<<x.first<<" "<<x.second<<endl;
        
        vector<vector<int>> ans;
        vector<int> v;
        for(auto x:w)
            if(w[x.first]!=-1)
                v.push_back(x.first);
        ans.push_back(v);
        v.clear();
        
        for(auto x:l)
            if(l[x.first]==1)
                v.push_back(x.first);
        ans.push_back(v);
        return ans;
        
    }
};


/*












*/

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        TreeMap<Integer, Integer> m = new TreeMap<>();
        
        for (int i = 0; i < matches.length; i++) {
            m.put(matches[i][1], m.getOrDefault(matches[i][1], 0) + 1);
        }
                
        TreeSet<Integer> winners = new TreeSet<>();
        for (int i = 0; i < matches.length; i++) {
            if (!m.containsKey(matches[i][0])) {
                winners.add(matches[i][0]);
            }
        }
        
        List<List<Integer>> ans = new ArrayList<>();
        
        List<Integer> v = new ArrayList<>();
        for (Integer x : winners) {
            v.add(x);
        }
        ans.add(new ArrayList<>(v));  // Add a copy of the list
        
        v.clear();  // Clear the list for the second part
        
        for (Map.Entry<Integer, Integer> entry : m.entrySet()) {
            if (entry.getValue() == 1) {
                v.add(entry.getKey());
            }
        }
        ans.add(new ArrayList<>(v));  // Add a copy of the list
        
        return ans;
    }
}
